pyshelf/cloud/factory.py

```diff
@@ -8,12 +9,14 @@ def __init__(self, config, logger):
         self.logger = logger
 
     def create_storage(self, bucket_name):
-        c = self.config
         # Although bucketName exists in the config provided it is not
         # required and is not used because we want the ability to change
         # buckets when we want.
-        if not c.get("buckets").get(bucket_name):
+        bucket_config = utils.get_bucket_config(self.config, bucket_name)
+        if bucket_config is None:
             self.logger.warning("Access keys for {0} are not in your config.".format(bucket_name))
             raise BucketConfigurationNotFound(bucket_name)
-        return Storage(c["buckets"][bucket_name]["accessKey"], c["buckets"][bucket_name]["secretKey"],
-                       bucket_name, self.logger)
+
+        storage = Storage(bucket_config["accessKey"], bucket_config["secretKey"], bucket_config["name"], self.logger)
```

Comment:
YOU DO NOT NEED TO FIX THIS! Also I don't think you need to force something to be on a single line. If it is more readable to do on separate lines then do it.  My previous comment (which I am sure some might disagree about it) is the formatting of multi line function calls.  In this case I would like to see...

```
storage = Storage(
    bucket_config["accessKey"],
    bucket_config["secretKey"],
    bucket_config["name"],
    self.logger
)
```

Its just a style/readability thing.  We can discuss it later.